cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,342 @@
+ */
+export function verifyQueryHitsText(expectedQueryHitsText) {
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsText);
+}
+
+/**
+ * Check if the Doc table's rowNumberth row's Filter For and Filter Out buttons exists.
+ * @param {number} rowNumber Doc table row number to check (First row is row 0)
+ * @param {boolean} shouldExist Should this button exist
+ */
+export function verifyDocTableRowFilterForAndOutButton(rowNumber, shouldExist) {
+  const shouldText = shouldExist ? 'exist' : 'not.exist';
+  getDocTableField(0, rowNumber).within(() => {
+    cy.getElementByTestId('filterForValue').should(shouldText);
+    cy.getElementByTestId('filterOutValue').should(shouldText);
+  });
+}
+
+/**
+ * Check the Doc Table rowNumberth row's Filter buttons filters the correct value.
+ * @param {number} rowNumber Doc table row number to check (First row is row 0)
+ * @param {string} filterElement data-test-sub element for filter.
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ * @param {boolean} shouldMatch boolean to determine if same rowNumber text should match after filter is applied
+ * @example verifyDocTableFilterAction(0, 'filterForValue', '10,000', '1', true)
+ */
+export function verifyDocTableFilterAction(
+  rowNumber,
+  filterElement,
+  expectedQueryHitsWithoutFilter,
+  expectedQueryHitsAfterFilterApplied,
+  shouldMatch
+) {
+  getDocTableField(0, rowNumber).then(($field) => {
+    const shouldText = shouldMatch ? 'have.text' : 'not.have.text';
+
+    const filterFieldText = $field.find('span span').text();
+    $field.find(`[data-test-subj="${filterElement}"]`).click();
+    // Verify pill text
+    cy.getElementByTestId('globalFilterLabelValue', {
+      timeout: 10000,
+    }).should('have.text', filterFieldText);
+    cy.getElementByTestId('discoverQueryHits').should(
+      'have.text',
+      expectedQueryHitsAfterFilterApplied
+    ); // checkQueryHitText must be in front of checking first line text to give time for DocTable to update.
+    getDocTableField(0, rowNumber).find('span span').should(shouldText, filterFieldText);
+  });
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+}
+
+/**
+ * Check if the first expanded Doc Table Field's first row's Filter For, Filter Out and Exists Filter buttons are disabled.
+ * @param {boolean} isEnabled Boolean determining if these buttons are disabled
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowFilterForFilterOutExistsFilterButtons(
```

Comment:
Why we need a utility to check enable or disable? We can write directly in the test, right?
```
it('verifies filter buttons state', () => {
  const shouldText = isEnabled ? 'be.enabled' : 'be.disabled';
  cy.getElementByTestId('addInclusiveFilterButton').should(shouldText);
  cy.getElementByTestId('removeInclusiveFilterButton').should(shouldText);
  cy.getElementByTestId('addExistsFilterButton').should(shouldText);
});
```